tests/bootstrap.php

```diff
@@ -1,54 +1,38 @@
 <?php
-$wgRunningUnitTests = true; // don't include DevBoxSettings when running unit tests
-$wgDevelEnvironment = true;
+/**
+ * This file handles initializing MediaWiki stack and setting global flags
+ * before handing over execution to PHPUnit that executes the tests
+ * provided in its configuration file set via the command line.
+ */
 
-class FakeCache {
-	public function __call($a, $b) {
-		return '';
-	}
-}
+error_reporting( E_ALL );
+ini_set( 'display_errors', 1 );
 
-require_once __DIR__ . '/../maintenance/commandLine.inc';
+// don't include DevBoxSettings when running unit tests
+$wgRunningUnitTests = true;
+$wgDevelEnvironment = true;
+$wgAnnotateTestSpeed = ( getenv( 'ANNOTATE_TEST_SPEED' ) === '1' );
 
-class MessageStub extends Message {
-	public function toString() {
-		throw new WikiaException('No messaging in unit tests');
-	}
-}
+require_once __DIR__ . '/../maintenance/Maintenance.php';
 
-function wfMsgExtStub() {
-	return '';
-}
-function wfMsgRealStub() {
-	return '';
-}
-function wfGetDBStub() {
-	throw new WikiaException('No DB in unit tests');
-}
+class PhpUnit extends Maintenance {
+	public function __construct() {
+		parent::__construct();
 
-function sno_callback($className) {
-	if ($className == 'Message') {
-		$className = 'MessageStub';
+		$this->addOption( 'slow-list', 'Whether to find and mark slow tests' );
 	}
 
-	return $className;
-}
-
-if ($wgRunningUnitTests && $wgNoDBUnits) {
-	rename_function('wfMsgExt', 'wfMsgExt_orig');
-	rename_function('wfMsgReal', 'wfMsgReal_orig');
-	rename_function('wfGetDB', 'wfGetDB_orig');
-	rename_function('wfMsgExtStub', 'wfMsgExt');
-	rename_function('wfMsgRealStub', 'wfMsgReal');
-	rename_function('wfGetDBStub', 'wfGetDB');
-
-	$wgMemc = new FakeCache();
-	$messageMemc = null;
-	$parserMemc = null;
+	public function execute() {
+		// TODO: Broken. Fix.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)